File: Practice/src/sort/KthNumber.js
function solution(array, commands) {
    return commands.map(
        (arr) =>
            array.slice(arr[0] - 1, arr[1]).sort((a, b) => a - b)[arr[2] - 1]
    );
    // var answer = [];
    // for (var i = 0; i < commands.length; i++) {
    //     answer.push(
    //         array
    //             .slice(commands[i][0] - 1, commands[i][1])
    //             .sort(function (a, b) {
    //                 return a - b;
    //             })[commands[i][2] - 1]
    //     );
    // }
    // return answer;
}

File: Practice/src/sort/KthNumber.java
package sort;

import java.util.Arrays;

public class KthNumber {
//	K번째수
	public int[] solution(int[] array, int[][] commands) {
		return Arrays.stream(commands)
				.mapToInt(i -> Arrays.stream(Arrays.copyOfRange(array, i[0] - 1, i[1])).sorted().toArray()[i[2] - 1])
				.toArray();
//		int[] temp = {};
//		int[] answer = new int[commands.length];
//		for (int i = 0; i < commands.length; i++) {
//			temp = Arrays.copyOfRange(array, commands[i][0] - 1, commands[i][1]);
//			Arrays.sort(temp);
//			answer[i] = temp[commands[i][2] - 1];
//		}
//		return answer;
	}
}
